pkg/sql/drop_index.go

```diff
@@ -135,6 +159,18 @@ func (p *planner) dropIndexByName(
 	if dropped {
 		return nil
 	}
+	// Check if a unique constraint depends on this index and raise an
+	// error if DropBehavior is DropCascade and the column is not being
+	// dropped.
+	if !droppingColumn && behavior != tree.DropCascade {
+		for name := range constraints {
+			if constraints[name].Kind == sqlbase.ConstraintTypeUnique &&
+				constraints[name].Index.ID == idx.ID {
+				return pgerror.NewErrorf(pgerror.CodeInvalidObjectDefinitionError,
+					"index %q is in use by unique constraint %q", idx.Name, name)
```

Comment:
The constraint name is just the index name, so don't need to print it twice. Could also mention CASCADE if we wanted to try to preempt having to look it up.